### PR TITLE
Bump to newer protoc versions

### PR DESCRIPTION
**Pull request by devinrsmith**
Unfortunately, I think some of the ts/js logic has changed, and I'm not able to plumb through the webpacking part on the deephaven-core side yet. Will need to chat with @niloc132 or @mofojed to try and get this sorted out.

go / python / java seem to be happy.

### CODE REVIEW

**Comment by devinrsmith** [comment]
Some of the logic for invoking protoc may have changed:

```
  /opt/protoc/bin/protoc \
    --plugin=protoc-gen-ts=/usr/src/app/node_modules/.bin/protoc-gen-ts \
    --js_out=import_style=commonjs:/generated/js \
    --ts_out=service=grpc-web:/generated/js \
    ...
```

is now complaining that it can't find protoc-gen-js. 

Explicitly adding the line:

```
  /opt/protoc/bin/protoc \
    --plugin=protoc-gen-ts=/usr/src/app/node_modules/.bin/protoc-gen-ts \
    --plugin=protoc-gen-js=/usr/src/app/node_modules/.bin/protoc-gen-ts \
    --js_out=import_style=commonjs:/generated/js \
    --ts_out=service=grpc-web:/generated/js \
```

produces a bunch of errors:

```
BrowserFlight_pb.d.ts: Tried to write the same file twice.
Flight_pb.d.ts: Tried to write the same file twice.
deephaven/proto/ticket_pb.d.ts: Tried to write the same file twice.
...
```

Commenting out the js_out part:

```
  /opt/protoc/bin/protoc \
    --plugin=protoc-gen-ts=/usr/src/app/node_modules/.bin/protoc-gen-ts \
    # --js_out=import_style=commonjs:/generated/js \
    --ts_out=service=grpc-web:/generated/js \
```

produces a variety of webpack errors:

```
...
ERROR in ./src/index.js 12:0-27
Module not found: Error: Can't resolve 'BrowserFlight_pb' in '/usr/src/app/raw-js-openapi/src'
...
```

Other similar variations resulted in similar errors.

**Comment by devinrsmith** [comment]
For reference, to build this image locally:

```bash
docker buildx bake --load protoc-base
```

And in deephaven-core, update `docker/registry/protoc-base/gradle.properties` to:

```
io.deephaven.project.ProjectType=DOCKER_REGISTRY
deephaven.registry.imageName=ghcr.io/deephaven/protoc-base:latest
deephaven.registry.imageId=ghcr.io/deephaven/protoc-base:latest
```

File: proto/01-protoc-base/install.sh
Comment (mofojed) [inline proto/01-protoc-base/install.sh:26]: Looks like there's an issue with v21 of protoc: https://github.com/protocolbuffers/protobuf-javascript/issues/127
Recommend downgrading to v20.1 (libprotoc 3.20.1)

Comment (devinrsmith) [reply proto/01-protoc-base/install.sh:26]: Thanks for the link, most likely the cause of errors I'm seeing.

**Comment by devinrsmith** [comment]
Updated with advice from @mofojed, things looking good wrt deephaven-core.

Desire to get this in for next release, see:

https://github.com/protocolbuffers/protobuf/security/advisories/GHSA-h4h5-3hr4-j3g2
https://github.com/protocolbuffers/protobuf/security/advisories/GHSA-wrvw-hg22-4m67

**Review by mofojed — APPROVED** [review]
Would wait for Colin's approval as well

**Review by niloc132 — APPROVED** [review]

**Pull request merged**
